proof/refine/AARCH64/InvariantUpdates_H.thy

```diff
@@ -5,11 +5,102 @@
  *)
 
 theory InvariantUpdates_H
-imports Invariants_H
+imports ArchInvsLemmas_H
 begin
 
+(* generic consequences which require arch-specific proofs *)
```

Comment:
Hmm, I understand that this is a convenient location due to the import graph, but these requalifies being put here does feel like a bit of a negative consequence of how we're doing this arch-split. They don't really have any connection with the rest of this file and feel pretty disconnected to `Invariants_H` and `ArchInvsLemmas_H`, which are the files that actually relate to them.

I don't have any better suggestions so am fine to leave them here for now, but I can imagine negative consequences going forwards, where we make changes or add new lemmas to `ArchInvsLemmas_H` and don't think to change here as well, resulting in a new, inconsistent approach.

Reply:
I agree it's not ideal. The issue is the second best place to put them is inside the Arch file, and then there's a risk of de-sync between arches (not that bad in general, but not fun when doing the arch-split itself).
Another way is to use locales, which is something that I'll be aiming for for the rest of the files, but for invariants the dependency graph was a bit much. Maybe I can add a comment to clarify or something?

Reply:
Ah right, I hadn't thought of this being specifically an artefact of the complicated invariants files.